test/should_fail/record_index.erl

```diff
@@ -1,9 +1,13 @@
 -module(record_index).
 
--export([f/0]).
+-export([f/0, g/0]).
 
 -record(rec, { apa :: integer()}).
 
 -spec f() -> boolean().
 f() ->
     #rec.apa.
+
+-spec g() -> 2.
+g() ->
+    #rec.apa.
```

Comment:
Why should this fail? `apa` is the first record element, i.e. the 2nd tuple element, isn't it?

Reply:
the inferred type if record index is currently `integer()`, so this check should fail because it is not a subtype of `2`. But you are right, I wanted to wright a testcase which will still fail when exact record index inference will be added... but I failed :)

Reply:
Move this one too known_problems/should pass and add a similar one with the wrong number!

Reply:
right, instead of creating a known problem, I pushed another commit with "very precise record index type", not to increase tech debt.